src/RageSoundMixBuffer.cpp

```diff
@@ -8,51 +8,37 @@
 
 RageSoundMixBuffer::RageSoundMixBuffer()
 {
-	m_pMixbuf = static_cast<float*>(std::malloc(BUF_SIZE * sizeof(float)));
-	if (m_pMixbuf == nullptr)
-	{
-		ASSERT_M(false, "Failed to allocate memory for the sound mixing buffer");
-	}
-	m_iBufSize = BUF_SIZE;
-	std::memset(m_pMixbuf, 0, m_iBufSize * sizeof(float));
+	m_iBufSize = 0;
 	m_iBufUsed = 0;
+	m_pMixbuf = nullptr;
 	m_iOffset = 0;
 }
 
-
 RageSoundMixBuffer::~RageSoundMixBuffer()
 {
 	std::free(m_pMixbuf);
 }
 
-/* write() will start mixing iOffset samples into the buffer.  Be careful; this is
- * measured in samples, not frames, so if the data is stereo, multiply by two. */
-void RageSoundMixBuffer::SetWriteOffset( int iOffset )
+void RageSoundMixBuffer::Extend(unsigned iSamples) noexcept
 {
-	m_iOffset = iOffset;
-}
+    const std::int_fast64_t realsize = static_cast<std::int_fast64_t>(iSamples) + static_cast<std::int_fast64_t>(m_iOffset);
```

Comment:
"std or not std, that is the question", I'm confused what to use when writting code..

Reply:
Argh.... This one got past me.  NO `std` prefix please!  

Thanks for catching this!